frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/MainActivity.java

```diff
@@ -306,6 +296,16 @@ private void handleIntent(Intent intent) {
             log("usb: detached");
             this.updateDevice();
         }
+        // Handle 'aopp:' URIs. This is called when the app is launched and also if it is already
+        // running and brought to the foreground.
+        if(Intent.ACTION_VIEW.equals(intent.getAction())) {
```

Comment:
strange to read. all the other branches have it reversed: `intent.getAction().equals(Intent.ACTION_VIEW)`

Reply:
also, might as well turn these `if`s into a single switch statement?

```java
switch (intent.getAction()) {
  case ACTION_USB_PERMISSION:
    ...
    break
  case UsbManager.ACTION_USB_DEVICE_ATTACHED:
  case UsbManager.ACTION_USB_DEVICE_DETACHED:
  case Intent.ACTION_VIEW:
}

Reply:
I'd be worried if `==` (switch) is the same as `.equals()`. I'll just reverse the order to make it consistent as you say.